data/js/tomahawk.js

```diff
@@ -508,14 +509,26 @@ Tomahawk.ajax = function(url, settings) {
 
             return str.join("&");
         };
+        if (typeof settings.data === 'object') {
+            if (settings.encoding == 'form') {
+                settings.data = formEncode(settings.data);
+                settings.content_type = settings.content_type || 'application/x-www-form-urlencoded';
+            } else if (settings.encoding == 'json') {
+                settings.data = JSON.stringify(settings.data);
+                settings.content_type = settings.content_type || 'application/json';
+            } else {
+                throw new Error("Tomahawk.ajax: unknown data encoding requested: " + settings.encoding);
+            }
+        }
 
         if (settings.type.toLowerCase() === 'get') {
-            settings.url += '?' + formEncode(settings.data);
+            settings.url += '?' + settings.data;
```

Comment:
Still not convinced, I believe it's easier to miss this and to cause trouble than it's useful.. I would rather not merge this part of the PR